x/market/client/cli/tx.go

```diff
@@ -0,0 +1,71 @@
+package cli
+
+import (
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+	"io/ioutil"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	// "github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/medibloc/panacea-core/v2/x/market/types"
+)
+
+// GetTxCmd returns the transaction commands for this module
+func GetTxCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      fmt.Sprintf("%s transactions subcommands", types.ModuleName),
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	cmd.AddCommand(NewCreateDealCmd())
+	return cmd
+}
+
+func NewBuildCreateDealMsg(clientCtx client.Context, txf tx.Factory, fs *flag.FlagSet) (tx.Factory, sdk.Msg, error) {
+	deal, err := parseCreateDealFlags(fs)
+	if err != nil {
+		return txf, nil, fmt.Errorf("faild to parse deal: %w", err)
+	}
+
+	budget, err := sdk.ParseCoinNormalized(deal.Budget)
+	if err != nil {
+		return txf, nil, err
+	}
+
+	msg := types.NewMsgCreateDeal(
+		deal.DataSchema,
+		&budget,
+		deal.WantDataCount,
+		deal.TrustedDataValidators,
+		clientCtx.GetFromAddress().String(),
+	)
+
+	return txf, msg, nil
+}
+
+func parseCreateDealFlags(fs *flag.FlagSet) (*createDealInputs, error) {
+	deal := &createDealInputs{}
+	dealFile, _ := fs.GetString(FlagDealFile)
+
+	if dealFile == "" {
+		return nil, fmt.Errorf("need deal json file using --%s flag", FlagDealFile)
+	}
+
+	contents, err := ioutil.ReadFile(dealFile)
```

Comment:
Does it mean the `CreateDealInput` is received as a file?
If then, we'd better receive it as args, not file.

Reply:
Yes, I've heard that when we create deal, we will use json file.

Reply:
I think it's no proble to receive data in a file. 
Because the [update of did](https://github.com/medibloc/panacea-core/blob/master/x/did/client/cli/txDID.go#L100) and [pool creation of osmosis](https://github.com/osmosis-labs/osmosis/blob/main/x/gamm/client/cli/tx.go#L46) is already received as a json file.

However,some tx uses both json file and argument.

I think it would be good to apply in both ways.(json file and argument) 😀